serpentTools/parsers/depletion.py

```diff
@@ -132,11 +135,19 @@ def _processChunk(self, chunk, name, variable):
         self.materials[name].addData(variable, cleaned)
 
     def _precheck(self):
-        """todo
+        """do a quick scan to see how many materials appear in the file
         """
-        pass
+        with open(self.filePath) as fh:
+            for line in fh:
+                sline = line.split()
+                if sline == []:
+                    continue
+                elif 'MAT' == line.split()[0]:
+                    self.materialCount += 1
 
     def _postcheck(self):
-        """todo
+        """ensure the parser grabbed the same number of counted materials
         """
-        pass
+        if len(self.materials) != self.materialCount:
+            error("Did not parse as many materials as counted in file\n"
+                  "{}".format(self.filePath))
```

Comment:
Replace with `if not self.materials`:

It is possible that the reader will not store the entire set of materials, as specified by the [`depletion.materials`](https://github.com/CORE-GATECH-GROUP/serpent-tools/blob/master/serpentTools/settings.py#L44) setting. It is acceptable to only select a single material, or if your incorrectly set this setting, you may not store any materials.

Reply:
This would make more sense in the precheck. There's not much I can fathom at the moment to have on the postcheck. Something reasonable would be checking day values indeed go out to the maximum burnup specified for that Serpent run in order to detect premature Serpent exits. Unlike with coef stuff, there's not a good way to get this for materials in advance, I think.